Common/Product/ReplWindow/Repl/ReplWindow.cs

```diff
@@ -2126,47 +2130,58 @@ private sealed class CommandFilter : IOleCommandTarget {
                     Debug.Assert(_outputBuffer.CurrentSnapshot.GetText(deleteSpan) == lineBreak);
                     edit.Delete(deleteSpan);
                     oldBufferLength -= lineBreak.Length;
+                    _addedLineBreakOnLastOutput = false;
                 }
 
-                edit.Insert(oldBufferLength, text);
-                if (lastOutput && !_readingStdIn && !EndsWithLineBreak(text)) {
+                var text = String.Empty;
+                var startPosition = oldBufferLength;
+                foreach (var entry in entries) {
+                    text = entry.Buffer.ToString();
+                    edit.Insert(oldBufferLength, text);
+
+                    var span = new Span(startPosition, text.Length);
+                    spans.Add(new KeyValuePair<Span, OutputBuffer.OutputEntryProperties>(span, entry.Properties));
+                    startPosition += text.Length;
+                }
+                if (!_readingStdIn && !EndsWithLineBreak(text)) {
                     var lineBreak = GetLineBreak();
                     edit.Insert(oldBufferLength, lineBreak);
-                    newOutputLength += lineBreak.Length;
                     _addedLineBreakOnLastOutput = true;
-                } else {
-                    _addedLineBreakOnLastOutput = false;
+                    // Adust last span.
```

Comment:
nit: this comment doesn't add much info - either make it more specific or remove